### PR TITLE
Topnavigation implementation

### PR DESCRIPTION
**Pull request by DarinaBlazevica**
- Add Topnavigtion component
- Add Topnavigation styling
- Add Cart menu component
- Add Menu component
- Outside click handler

### CODE REVIEW

File: src/components/organisms/Topnavigation/Topnavigation.tsx [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+  cartMenu: CartMenuData;
+  menu: MenuData[];
+}
+
+const TopNavigation = (props: TopnavigationProps) => {
+  const topnavigationProps = props;
+
+  const [isCartMenuOpen, setIsCartMenuOpen] = useState<boolean>(false);
+  const [isMenuOpen, setIsMenuOpen] = useState<boolean>(false);
+  const [hamburgerStyle, setHamburgerStyle] = useState<string>("bar");
```
Comment (ghost) [inline src/components/organisms/Topnavigation/Topnavigation.tsx]: Create enum for hamburger style and use that

File: src/components/atoms/buttons/Hamburger.tsx [hunk 2]
```diff
@@ -0,0 +1,19 @@
+import "./Hamburger.css";
+
+interface Hamburger {
```
Comment (ghost) [inline src/components/atoms/buttons/Hamburger.tsx]: HamburgerProps

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/components/organisms/Topnavigation/Topnavigation.tsx (hunk 1 above)
Comment (DarinaBlazevica) [reply src/components/organisms/Topnavigation/Topnavigation.tsx]: Added

File: src/components/atoms/buttons/Hamburger.tsx (hunk 2 above)
Comment (DarinaBlazevica) [reply src/components/atoms/buttons/Hamburger.tsx]: Changed

**Review by ghost — APPROVED** [review]

**Pull request merged**